### ISSUE TITLE
Performance problem with creating multiple pages

### ISSUE THREAD

**Issue by hulinNeil** [diegomura/react-pdf#2448]
**Describe the bug**
I tried to create multiple pages pdf in web worker,  and found that the speeds decreased significantly as the number increased.

Graphs of speed and data count:
<img width="441" alt="image" src="https://github.com/diegomura/react-pdf/assets/24296884/2af83cf8-7b6f-4798-9442-3f911de73972">

|Data count       |Time         |Speeds  (count / time)          |
| -------------|:--------------:|:--------------:|
|10|string|Y|
|30|string|Y|

**To Reproduce**
Use the online [demo](https://stackblitz.com/edit/vite-react-ts-5wmsap) to check the problem.
Click on different buttons to see different speeds.
![image](https://github.com/diegomura/react-pdf/assets/24296884/c0defe66-8c6c-4c26-985b-edafc6b122a2)

**Expected behavior**
Expect to generate at a constant rate, or faster on multiple pages.

**Desktop (please complete the following information):**
 - OS: MacOS
 - Browser: Chrome
 - React-pdf version : 3.1.13

**Comment by diegomura** [comment]
I'm constantly trying to enhance performance, but PDF generation is not cheap and layout computation either. I can leave this open but there's no obvious solution we havent done already

**Comment by CodeFromAnywhere** [comment]
Curious about this. I have used `useMemo` hook to only render the `<Document/>` once but it still seems to make the UI laggy. Would it be possible to use a Worker somehow, not blocking the UI thread while doing PDF calculations in the worker?

**Comment by Chachaproper** [comment]
> Curious about this. I have used `useMemo` hook to only render the `<Document/>` once but it still seems to make the UI laggy. Would it be possible to use a Worker somehow, not blocking the UI thread while doing PDF calculations in the worker?

Yes, the documentation says this. On our project we did this until we transferred the entire render to the server.
